[CI] ClassificationHousePricingIT testFeatureImportanceValues failing

**Build Scans:**
- [elasticsearch-intake #15766 / part4](https://gradle-enterprise.elastic.co/s/jos4xy3r5j5j6)
- [elasticsearch-periodic #5850 / openjdk17_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/clfmgzbq3baii)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest" --tests "org.elasticsearch.xpack.ml.integration.ClassificationHousePricingIT.testFeatureImportanceValues" -Dtests.seed=1572DE2E38E566FA -Dtests.locale=so-Latn-SO -Dtests.timezone=Europe/Andorra -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ml.integration.ClassificationHousePricingIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testFeatureImportanceValues),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Stats were: {"id":"classification_house_pricing_test_feature_importance_values","state":"started","progress":[{"phase":"reindexing","progress_percent":100},{"phase":"loading_data","progress_percent":100},{"phase":"feature_selection","progress_percent":100},{"phase":"coarse_parameter_search","progress_percent":100},{"phase":"fine_tuning_parameters","progress_percent":100},{"phase":"final_training","progress_percent":100},{"phase":"writing_results","progress_percent":100},{"phase":"inference","progress_percent":70}],"data_counts":{"training_docs_count":510,"test_docs_count":669,"skipped_docs_count":0},"memory_usage":{"timestamp":1736895796099,"peak_usage_bytes":572835,"status":"ok"},"analysis_stats":{"classification_stats":{"timestamp":1736895796099,"iteration":0,"hyperparameters":{"class_assignment_objective":"maximize_minimum_recall","alpha":0.05847558895962435,"downsample_factor":0.23752714673596464,"eta":0.02070965477416433,"eta_growth_rate_per_tree":1.01035
[truncated]
```

**Issue Reasons:**
- [8.x] 2 failures in test testFeatureImportanceValues (1.4% fail rate in 146 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -385,9 +385,6 @@ tests:
 - class: org.elasticsearch.oldrepos.OldRepositoryAccessIT
   method: testOldSourceOnlyRepoAccess
   issue: https://github.com/elastic/elasticsearch/issues/120080
-- class: org.elasticsearch.xpack.ml.integration.ClassificationHousePricingIT
-  method: testFeatureImportanceValues
-  issue: https://github.com/elastic/elasticsearch/issues/116564
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
   method: test {date_nanos.Bucket Date nanos by 10 minutes}
   issue: https://github.com/elastic/elasticsearch/issues/120162
@@ -504,9 +501,6 @@ tests:
 - class: org.elasticsearch.xpack.security.profile.ProfileIntegTests
   method: testHasPrivileges
   issue: https://github.com/elastic/elasticsearch/issues/121346
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsTextAndKeyword
-  issue: https://github.com/elastic/elasticsearch/issues/121502
 - class: org.elasticsearch.xpack.security.profile.ProfileIntegTests
   method: testSuggestProfilesWithName
   issue: https://github.com/elastic/elasticsearch/issues/121022

